landingai/storage/data_access.py

```diff
@@ -0,0 +1,71 @@
+import logging
+import tempfile
+from pathlib import Path
+from typing import Optional
+from urllib.parse import urlparse
+
+from typing import Optional
+
+
+from landingai.io import read_file
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def download_public_file(
```

Comment:
i was thinking file downloading is `io` module's responsibility, i.e. handling io operations.
but i'd love to know your thoughts here.

Regardless, i think it's beneficial to a define the responsibility of the `data_access` module versus the `io` module explicitly with doc string at the top of the module file.

Reply:
nit, about naming, how about just "download_file()`?

Reply:
it's worth a unit test.

Reply:
Added test